app/models.py

```diff
@@ -3075,16 +3082,55 @@ class Partner(Base, ModelMixin):
     name = sa.Column(sa.String(128), unique=True, nullable=False)
     contact_email = sa.Column(sa.String(128), unique=True, nullable=False)
 
+    @staticmethod
+    def find_by_token(token: str) -> Optional[Partner]:
+        hmaced = PartnerApiToken.hmac_token(token)
+        res = (
+            Session.query(Partner, PartnerApiToken)
+            .filter(
+                and_(
+                    PartnerApiToken.token == hmaced,
+                    Partner.id == PartnerApiToken.partner_id,
+                )
+            )
+            .first()
+        )
+        if res is None or res.Partner is None:
```

Comment:
Oh I didn't know we can use `res.Partner` here. Usually the result is unpacked with `partner, api_token=res`

Reply:
Changed it

Reply:
Sorry, re-reading the comment I have noticed I misunderstood. Fixed the `None` crash and replaced it with the destructuring syntax

Reply:
Either keep it as it is or check for res not to be None before unpack. Otherwise it will fail :)